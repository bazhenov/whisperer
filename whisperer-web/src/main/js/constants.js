export const SSE_RECONNECT_TIMEOUT = 1000;
export const LS_CONNECTION_PARAMS_KEY = 'connectionParams';
export const MAX_MESSAGES_COUNT = 50000;
export const MAX_MESSAGES_TO_DISPLAY = 1000;