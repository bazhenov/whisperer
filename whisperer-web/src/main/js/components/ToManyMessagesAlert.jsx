import React from "react";
import { MAX_MESSAGES_TO_DISPLAY } from '../constants'

export default ({ filteredCount }) => {
	if (filteredCount < MAX_MESSAGES_TO_DISPLAY) return null;
	return <div className="row">
		<div className="col-md-12">
			<div className="alert alert-warning" role="alert">
				Too many messages! Only first {MAX_MESSAGES_TO_DISPLAY} was showed.
			</div>
		</div>
	</div>
}