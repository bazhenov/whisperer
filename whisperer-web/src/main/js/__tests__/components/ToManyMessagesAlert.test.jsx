import React from "react";
import ToManyMessagesAlert from "../../components/ToManyMessagesAlert.jsx";
import renderer from "react-test-renderer";
import { MAX_MESSAGES_TO_DISPLAY } from '../../constants'

it('ToManyMessagesAlert', () => {
	let tree = renderer.create(
		<ToManyMessagesAlert filteredCount={0}/>
	).toJSON();
	expect(tree).toMatchSnapshot();

	tree = renderer.create(
		<ToManyMessagesAlert filteredCount={MAX_MESSAGES_TO_DISPLAY}/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});