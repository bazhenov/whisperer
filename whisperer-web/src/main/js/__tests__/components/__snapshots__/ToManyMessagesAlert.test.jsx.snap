// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToManyMessagesAlert 1`] = `null`;

exports[`ToManyMessagesAlert 2`] = `
<div
  className="row"
>
  <div
    className="col-md-12"
  >
    <div
      className="alert alert-warning"
      role="alert"
    >
      Too many messages! Only first 
      1000
       was showed.
    </div>
  </div>
</div>
`;
